graphql-dgs-reactive/src/main/kotlin/com/netflix/graphql/dgs/reactive/internal/DefaultDgsReactiveQueryExecutor.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.reactive.internal
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.jayway.jsonpath.DocumentContext
+import com.jayway.jsonpath.JsonPath
+import com.jayway.jsonpath.TypeRef
+import com.jayway.jsonpath.spi.mapper.MappingException
+import com.netflix.graphql.dgs.exceptions.DgsQueryExecutionDataExtractionException
+import com.netflix.graphql.dgs.exceptions.QueryException
+import com.netflix.graphql.dgs.internal.*
+import graphql.ExecutionResult
+import graphql.execution.ExecutionIdProvider
+import graphql.execution.ExecutionStrategy
+import graphql.execution.NonNullableFieldWasNullError
+import graphql.execution.instrumentation.ChainedInstrumentation
+import graphql.schema.GraphQLSchema
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpHeaders
+import org.springframework.web.reactive.function.server.ServerRequest
+import reactor.core.publisher.Mono
+import java.util.*
+import java.util.concurrent.atomic.AtomicReference
+
+class DefaultDgsReactiveQueryExecutor(
+    defaultSchema: GraphQLSchema,
+    private val schemaProvider: DgsSchemaProvider,
+    private val dataLoaderProvider: DgsDataLoaderProvider,
+    private val contextBuilder: DefaultDgsReactiveGraphQLContextBuilder,
+    private val chainedInstrumentation: ChainedInstrumentation,
+    private val queryExecutionStrategy: ExecutionStrategy,
+    private val mutationExecutionStrategy: ExecutionStrategy,
+    private val idProvider: Optional<ExecutionIdProvider>,
+    private val reloadIndicator: DefaultDgsQueryExecutor.ReloadSchemaIndicator = DefaultDgsQueryExecutor.ReloadSchemaIndicator { false }
+) : com.netflix.graphql.dgs.reactive.DgsReactiveQueryExecutor {
+    val logger: Logger = LoggerFactory.getLogger(DefaultDgsQueryExecutor::class.java)
+
+    val schema = AtomicReference(defaultSchema)
+
+    override fun execute(
+        query: String,
+        variables: MutableMap<String, Any>?,
+        extensions: MutableMap<String, Any>?,
+        headers: HttpHeaders?,
+        operationName: String?,
+        serverHttpRequest: ServerRequest?
+    ): Mono<ExecutionResult> {
+        val graphQLSchema: GraphQLSchema =
```

Comment:
Why not do this in one chain? Then all actions will be done not in Assembly time
Like this:
```
Mono.fromCallable {
        if (reloadIndicator.reloadSchema()). schema.updateAndGet { schemaProvider.schema() }
        else schema.get()
}
.zipWith( contextBuilder.build(DgsReactiveRequestData(extensions, headers, serverHttpRequest)) )
.flatMap { 
    val graphQlSchema = it.t1
    val dsgContext = it.t2
    Mono.fromCompletionStage(...)
}
```

Reply:
fixed